enforcer/datapath/datapath.go

```diff
@@ -216,32 +215,40 @@ func NewWithDefaults(
 // Enforce implements the Enforce interface method and configures the data path for a new PU
 func (d *Datapath) Enforce(contextID string, puInfo *policy.PUInfo) error {
 
-	puContext, err := d.contextTracker.Get(contextID)
-	if err != nil {
-		// Call proxy enforce from here and not from trireme like for other calls
-		// We will call enforce every time on the enforce so no need to propagate this info out of enforcer package
-		zap.L().Debug("Called Proxy Enforce")
-
-		proxyerr := d.proxyhdl.Enforce(contextID, puInfo)
-		if proxyerr != nil {
-			zap.L().Error("Unable to Enforce", zap.Error(proxyerr))
-			return proxyerr
-		}
-
-		createerr := d.doCreatePU(contextID, puInfo)
-		if createerr != nil {
-			zap.L().Error("Called Do Create Returned error", zap.Error(createerr))
-			return createerr
-		}
+	zap.L().Debug("Called Proxy Enforce")
 
-		return nil
-	}
-	//In case of update
+	// setup proxy before creating PU
 	proxyerr := d.proxyhdl.Enforce(contextID, puInfo)
```

Comment:
convert this to the form 

if err := xxx; err != nil {
     return err .  // OR return fmt.Errorf("Enforce failed: %s", err)
}

When you tweak code make this minor changes to fit our guidelines.